sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/Utility.java

```diff
@@ -280,4 +280,21 @@ public static OffsetDateTime parseDate(String dateString) {
                 });
         });
     }
+
+    /**
+     * Appends a query parameter to a url.
+     *
+     * @param url The url.
+     * @param key The query key.
+     * @param value The query value.
+     * @return The updated url.
+     */
+    public static String appendQueryParameter(String url, String key, String value) {
```

Comment:
Do we want to eagerly promote this in public API given it's only use is in Blob Batch?

Reply:
It's also used in Blob's getBlobUrl. That was actually where the method originally existed and I wanted to pull it out to prevent duplication

Reply:
Given there are more use cases for this sounds good to add.